### ISSUE TITLE
[GraphIR] Support non-flattened SoftMax in IR

### ISSUE THREAD

**Issue by aturetsk** [pytorch/glow#4462]
Currently Glow does tensor flattening in ONNX importer, as described in https://github.com/onnx/onnx/blob/master/docs/Operators.md#Softmax. However, object detection networks often have SoftMax nodes on tensors with width/height > 1, thus flattened tensor dimensions would be like [N * W * H, C]. I.e. information about exact dimension sizes, which might be important for a backend, is lost.
Can we have non-flattened (i.e. having arbitrary number of dimensions and "axis" parameter) SoftMax in Glow IR? The flattening can be moved from importer to the lowering stage, thus minimal fixes in existing backends will be required to adjust to the change.
We're looking at making this change in our local repo. Would be good to hear other people's opinion about this and, if possible, agree on something which could be upstreamable.

**Comment by aturetsk** [comment]
@ayermolo

**Comment by jfix71** [comment]
SGTM to support this. To be clear, are you suggesting that we add `axis` to Glow's `SoftMaxNode`, and then during lowering we lower e.g. a `SoftMax(axis=2)` to `SoftMax(axis=1)` with flattening done accordingly? Then a backend can prevent this lowering if it prefers to see the axis and handle it differently?

**Comment by aturetsk** [comment]
@jfix71, yes, I suggest exactly what you described.

**Comment by ayermolo** [comment]
@aturetsk As we discussed. I'll make the change.
